### PR TITLE
chore: update CRDs and api repo with new kubebuilder

### PR DESCRIPTION
**Pull request by MatousJobanek**
result of codeready-toolchain/api#237
* CRDs use a new name format: `toolchain.dev.openshift.com_<crd-plural-name>.yaml`
* sample CRs renamed to use the new name format as well
* slightly modified generated CRD as a result of using new kubebuilder (controller-tools)
* small changes in go.mod, so we can use the new controller-tools 0.4.1 together with client-go 0.18.3
* update the imports to the api repo/types

### CODE REVIEW

File: cmd/manager/main.go [hunk 1]
```diff
@@ -9,7 +9,7 @@ import (
 	"runtime"
 	"strings"
 
-	api "github.com/codeready-toolchain/api/pkg/apis"
+	api "github.com/codeready-toolchain/api/api/v1alpha1"
```
Comment (xcoulon) [inline cmd/manager/main.go:12]: same comment as in https://github.com/codeready-toolchain/toolchain-common/pull/177#discussion_r636104231 regarding the alias for the `github.com/codeready-toolchain/api/api/v1alpha1` package

File: pkg/apis/apis.go [hunk 2]
```diff
@@ -1,7 +1,7 @@
 package apis
 
 import (
-	"github.com/codeready-toolchain/api/pkg/apis"
+	api "github.com/codeready-toolchain/api/api/v1alpha1"
```
Comment (xcoulon) [inline pkg/apis/apis.go:4]: same here, let's use `toolchainv1alpha1` if possible

**Review by xcoulon — APPROVED** [review]
Thanks for undertaking these changes @MatousJobanek 🙌 
I've left a comment with regards to the alias for our `github.com/codeready-toolchain/api/api/v1alpha1` package (as I did in other PRs), but it's not something we need to address in the PR.

**Review by alexeykazakov — APPROVED** [review]

File: cmd/manager/main.go (hunk 1 above)
Comment (MatousJobanek) [reply cmd/manager/main.go]: I already did that locally yesterday, but forgot to push the changes to GH: b8f6c51

**Pull request merged**